commons/src/test/java/io/aiven/kafka/tiered/storage/commons/chunkindex/serde/ChunkSizesBinaryCodecTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.chunkindex.serde;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class ChunkSizesBinaryCodecTest {
+    @Test
+    void empty() {
+        final List<Integer> values = List.of();
+        final var encoded = ChunkSizesBinaryCodec.encode(values);
+        final var decoded = ChunkSizesBinaryCodec.decode(encoded);
+        assertThat(decoded).isEqualTo(values);
+    }
+
+    @Test
+    void singleValue() {
+        final List<Integer> values = List.of(213);
+        final var encoded = ChunkSizesBinaryCodec.encode(values);
+        final var decoded = ChunkSizesBinaryCodec.decode(encoded);
+        assertThat(decoded).isEqualTo(values);
+    }
+
+    @Test
+    void singleMaxInt() {
+        final List<Integer> values = List.of(Integer.MAX_VALUE);
+        final var encoded = ChunkSizesBinaryCodec.encode(values);
+        final var decoded = ChunkSizesBinaryCodec.decode(encoded);
+        assertThat(decoded).isEqualTo(values);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideMultipleValues")
+    void multipleValues(final List<Integer> values) {
+        final var encoded = ChunkSizesBinaryCodec.encode(values);
+        final var decoded = ChunkSizesBinaryCodec.decode(encoded);
+        assertThat(decoded).isEqualTo(values);
+    }
+
+    static Stream<List<Integer>> provideMultipleValues() {
+        final List<Integer> longList = new ArrayList<>();
+        for (int i = 0; i < Integer.MAX_VALUE - 2000; i += 1000) {
+            longList.add(i);
+        }
+        final List<Integer> longRevertedList = new ArrayList<>(longList);
+        Collections.reverse(longRevertedList);
+        return Stream.of(
+            List.of(0, 1000, 2, 44002, 369),
+            List.of(Integer.MAX_VALUE, Integer.MAX_VALUE - 1, Integer.MAX_VALUE - 2, 10),
+            longList,
+            longRevertedList
```

Comment:
Nit: I would add more values here to cover all cases in `bytesNeeded`.

Reply:
Done